Makefile

```diff
@@ -29,43 +29,43 @@ $(THRIFT_BUILDDIR)/tests/integration/test.thrift_buildstamp: tests/integration/t
 	rm -f tests/integration/test_thrift/TestService-remote
 	touch $@
 
+.PHONY: venv
+venv:
+	poetry install --all-extras
+
 .PHONY: docs
 docs:
-	sphinx-build -M html docs/ build/
+	poetry run sphinx-build -M html docs/ build/
 
 .PHONY: doctest
 doctest:
-	sphinx-build -M doctest docs/ build/
+	poetry run sphinx-build -M doctest docs/ build/
 
 .PHONY: linkcheck
 linkcheck:
-	sphinx-build -M linkcheck docs/ build/
+	poetry run sphinx-build -M linkcheck docs/ build/
```

Comment:
I'm a little mixed on this approach.

The options here are:

* Prefix commands with `poetry run` like this. This always works but is a little noisy.
* Don't prefix commands with `poetry run`. This would require users to manually activate their virtualenv to run the make targets (or we could use direnv to activate it automatically, but this won't have universal adoption).
* Don't prefix commands with `poetry run`, but mess with the PATH in the Makefile itself. This probably works but feels a bit sloppy to me (probably need to unset it for the `make venv` call, for example).

Reply:
My preference is _always_ to have poetry make the venv in the project and never depend on poetry in scripts. instead doing `.venv/bin/python -m COMMAND` you can configure this in poetry.toml, env vars, or pyproject.

Reply:
++ I like that better, not sure why I neglected that option.